### PR TITLE
Fix: Casting & sync skills

### PR DESCRIPTION
**Pull request by Sparkymod**
- Fix: #310 
- Adding to: #67 
- Adding more mode for `SkillDamagePacket`
- Adding Packets: `SkillSync`
- Fixing packets: `SkillUse`
- Broadcasting packets: `RegionSkills`, `BuffSkills`
- Minor fixes.

Also thanks to @Zintixx for the help with the SyncSkillPacket

### CODE REVIEW

**Review by AngeloTadeucci — APPROVED** [review]

File: MapleServer2/PacketHandlers/Game/SkillHandler.cs [hunk 1]
```diff
@@ -43,96 +43,137 @@ public override void Handle(GameSession session, PacketReader packet)
...
+            byte motionPoint = packet.ReadByte();
+            CoordF position = packet.Read<CoordF>();
+            CoordF unkCoords = packet.Read<CoordF>();
+            CoordF rotation = packet.Read<CoordF>();
+            CoordF unknown = packet.Read<CoordF>();
+            bool toggle = packet.ReadBool();
             packet.ReadInt();
+            packet.ReadByte();
+
+            session.FieldManager.BroadcastPacket(SkillSyncPacket.Sync(skillSN, session.FieldPlayer, position, rotation, toggle));
```
Comment (Zintixx) [inline MapleServer2/PacketHandlers/Game/SkillHandler.cs:128]: This packet shouldn't be broadcasted to the skill caster.

File: MapleServer2/PacketHandlers/Game/SkillHandler.cs [hunk 2]
```diff
@@ -125,7 +125,7 @@ private static void HandleSyncSkills(GameSession session, PacketReader packet)
             packet.ReadInt();
             packet.ReadByte();
 
-            session.FieldManager.BroadcastPacket(SkillSyncPacket.Sync(skillSN, session.FieldPlayer, position, rotation, toggle));
+            session.Send(SkillSyncPacket.Sync(skillSN, session.FieldPlayer, position, rotation, toggle));
```
Comment (Zintixx) [inline MapleServer2/PacketHandlers/Game/SkillHandler.cs:128]: wait no I meant it should be:
`session.FieldManager.BroadcastPacket(SkillSyncPacket.Sync(skillSN, session.FieldPlayer, position, rotation, toggle), session);`
So every player would get it except the caster

Comment (Sparkymod) [reply MapleServer2/PacketHandlers/Game/SkillHandler.cs]: oh ok

**Review by Zintixx — APPROVED** [review]

File: MapleServer2/Packets/SkillDamagePacket.cs [hunk 3]
```diff
@@ -9,27 +9,64 @@ public static class SkillDamagePacket
...
+            pWriter.WriteInt(player.ObjectId);
+            pWriter.WriteInt(skillCast.SkillId);
+            pWriter.WriteShort(skillCast.SkillLevel);
+            pWriter.WriteByte(skillCast.MotionPoint);
+            pWriter.WriteByte(skillCast.AttackPoint);
+            pWriter.Write(position.ToShort());
+            pWriter.Write(rotation);
+            pWriter.WriteByte();
+            pWriter.WriteInt(skillCast.ServerTick);
+            pWriter.WriteByte(count); // Loop
```
Comment (AlanMorel) [inline MapleServer2/Packets/SkillDamagePacket.cs:38]: Unnecessary comment

File: MapleServer2/Packets/SkillDamagePacket.cs [hunk 4]
```diff
@@ -59,5 +111,90 @@ public static Packet ApplyHeal(Status status, int healAmount)
...
+            pWriter.WriteInt(unkInt);
+            pWriter.WriteByte();
+            pWriter.WriteByte(count);
+            for (int i = 0; i < count; i++)
+            {
+                pWriter.WriteInt(player.ObjectId);
+                pWriter.WriteByte(count2);
+                pWriter.Write(blockPosition);
+                pWriter.Write(direction);
+                for (int x = 0; x < count2; x++)
```
Comment (AlanMorel) [inline MapleServer2/Packets/SkillDamagePacket.cs:131]: Use `j` instead of `x`

File: MapleServer2/Packets/SkillDamagePacket.cs [hunk 5]
```diff
@@ -59,5 +111,90 @@ public static Packet ApplyHeal(Status status, int healAmount)
...
+            pWriter.WriteInt(skillCast.SkillId);
+            pWriter.WriteShort(skillCast.SkillLevel);
+            pWriter.WriteByte(targetCount);
+            for (int i = 0; i < targetCount; i++)
+            {
+                pWriter.WriteInt(player.ObjectId);
+                pWriter.WriteByte(count2);
+                pWriter.Write(position);
+                pWriter.Write(direction);
+                for (int x = 0; x < count2; x++)
```
Comment (AlanMorel) [inline MapleServer2/Packets/SkillDamagePacket.cs:157]: Same here

**Review by AlanMorel — CHANGES_REQUESTED** [review]

**Pull request merged**